pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -184,6 +184,27 @@ func (c *controller) checkReadyForEvents(curr interface{}) error {
 	return nil
 }
 
+func (c *controller) tryLedgerPut(obj interface{}, schema collection.Schema) {
+	iobj := obj.(crd.IstioObject)
+	om := iobj.GetObjectMeta()
+	key := model.Key(schema.Resource().Kind(), om.Name, om.Namespace)
+	_, err := c.client.configLedger.Put(key, om.ResourceVersion)
+	if err != nil {
+		scope.Errorf("Failed to update %s in ledger, status will be out of date.", key)
+	}
+}
+
+func (c *controller) tryLedgerDelete(obj interface{}, schema collection.Schema) {
+	iobj := obj.(crd.IstioObject)
+	om := iobj.GetObjectMeta()
+	key := model.Key(schema.Resource().Kind(), om.Name, om.Namespace)
+	err := c.GetLedger().Delete(key)
+	if err != nil {
+		scope.Errorf("Failed to delete %s in ledger, status will be out of date.", key)
```

Comment:
Why would this happen?

Reply:
It should never happen, but it is always possible to encounter an error when attempting to delete a key from a merkle tree.  Usually this would mean that a node was improperly expired, or that we had some similar bug in the merkle tree implementation.